internal/providers/kubernetes.go

```diff
@@ -0,0 +1,91 @@
+package providers
+
+import (
+	"context"
+	"log"
+	"os"
+	"path/filepath"
+
+	authv1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+const (
+	KubeconfigDefaultDir  = ".kube"
+	KubeconfigDefaultFile = "config"
+	StratusK8sNamespace   = "stratus-red-team"
+)
+
+type K8sProvider struct {
+	k8sClient *kubernetes.Clientset
+}
+
+var (
+	k8sProvider    K8sProvider
+	kubeConfigPath string
+)
+
+func K8s() *K8sProvider {
+	return &k8sProvider
+}
+
+// GetKubeConfigPath returns the path of the kubeconfig, looking at environment variables, and then
+// the home directory
+func GetKubeConfigPath() string {
+	if kubeConfigPath != "" {
+		return kubeConfigPath
+	}
+
+	// Set to default directory if environment variable is not set, ignore if file doesn't exist
+	// to default to in-cluster client config
+	if kubeConfigPath = os.Getenv("KUBECONFIG"); kubeConfigPath == "" {
```

Comment:
This works because `kubeConfigPath` is a var, but it feels confusing to read because at first look it looks like `kubeConfigPath` is scoped to the condition block